var/spack/repos/builtin/packages/texinfo/package.py

```diff
@@ -69,6 +69,13 @@ def setup_build_environment(self, env):
 
     @classmethod
     def determine_version(cls, exe):
+        # On CentOS and Ubuntu, the OS package texinfo installs "info",
+        # which satisfies spack external find, but "makeinfo" comes from
+        # texi2html and may not be installed (and vice versa).
+        info = which("info")
+        makeinfo = which("makeinfo")
+        if info is None or makeinfo is None:
+            return None
```

Comment:
A couple things:

* We may not modify PATH while searching prefixes for externals (these may differ from the searched paths when using `spack external find --path ...`)
  * I recommend instead decomposing `exe` into its base directory and looking inside of that
* Given that `info` is the executable we are looking for, we will know we have it here (i.e. `exe` will be an instance of `info`) so IMO it's unnecessary to force that

In the description of https://github.com/spack/spack/issues/40418, I got the impression that `texinfo` sometimes has `makeinfo` but always has `info`. If that's true and we only want `texinfo` instances that provide `makeinfo`, could we update like

```
executables = ["^makeinfo$"]
```

? (this would also possibly involve updating `determine_version` if `makeinfo` has different output). If I got that wrong, and `texinfo` installations could be missing either, then this suggestion won't work.

Reply:
I am not entirely sure of the last aspect - I think you can install `texi2html` (providing `makeinfo`) without installing `texlive` (providing `info`) and vice versa.

Reply:
I have to admit that this is confusing, because it seems to depend on the OS and version. Let me dig a little more, maybe we can just look for `makeinfo` after all and be done with it.

Reply:
FWIW it's fine to discard the suggestion to check for makeinfo if it isn't immediately clear whether it `info` would always be available (no need to survey various installations or look in obscure spots to see if there is some guarantee of this). In that case, I'd say just extract a sensible search dir from `exe`, look for `makeinfo` there, and bail if it doesn't exist.

Reply:
I did some research on Ubuntu 20, and it looks like `info` is coming from another package, and `makeinfo` is coming from `texinfo`. So I am not sure why `texinfo` was looking for just `info` in the first place, and I also don't know where I got the `texi2html` info from (apologies for that part, will correct the original issue).

Reply:
I just updated the issue as follows:

(Updated version after doing a few more checks)

On Ubuntu 20, the OS package `texinfo` installs `makeinfo`, which satisfies `spack external find`, but the executable `info` comes from package `info` and may not be installed (and vice versa).

For spack builds of `texinfo`, there is  logic to check that both `info` and `makeinfo` are built, but apparently this isn’t used for external packages:
```
sanity_check_is_file = [join_path("bin", "info"), join_path("bin", "makeinfo")]
```
I don't know why the current `texinfo` package in spack tries to look for both - are there situations where both are needed? If that is true, then `spack external find` should only find `texinfo` if both `info` and `makeinfo` are installed. And apparently the spack package `texinfo` does create both executables. However, as the package names on Ubuntu suggest, `info` is not part of package `texinfo`. Very confusing.

Reply:
It could be that some dependents can get away with one or the other, but that whoever wrote the Spack `texinfo` package decided it was easier to always bundle them together (and not for example to offer them as variants). If the Spack build insists on them being together, I think the external finding logic should insist on that too, and therefore that your solution is the right approach (although I still think this needs to change how it looks for `makeinfo`).

Reply:
@scheibelp Done, please check. Thanks!